### PR TITLE
Add support to postfix service for regexp-based alias files

### PR DESCRIPTION
**Pull request by ryantrinkle**

### CODE REVIEW

File: nixos/modules/services/mail/postfix.nix [hunk 1]
```diff
@@ -269,6 +273,14 @@ in
         ";
       };
 
+      aliasMapType = mkOption {
+        default = "hash";
+	type = lib.types.enum [ "hash" "regexp" ];
```
Comment (lethalman) [inline nixos/modules/services/mail/postfix.nix]: Wrong indent here.

**Comment by ryantrinkle** [comment]
Hi @lethalman; I addressed the spacing issue you mentioned, plus one other place where hard tabs needed to be replaced with spaces.

**Comment by rasendubi** [comment]
(triage) still an issue.

**Comment by Nadrieril** [comment]
(triage) Still an issue ? If noone is working on this PR anymore, please close it.

**Comment by uwap** [comment]
Looks good. Could you do it for virtual aliases too?

**Comment by rasendubi** [comment]
I am closing this PR because it got stuck. If anyone would like to implement this feature/fix merge conflicts, feel free to submit new PR.

**Pull request closed without merging**